examples/web-chat/src/hooks.ts

```diff
@@ -0,0 +1,90 @@
+import React, { useEffect, useState } from "react";
+import { generate } from "server-name-generator";
+import { Message } from "./Message";
+import { Decoder } from "@waku/core/lib/message/version_0";
+import { LightNode, StoreQueryOptions } from "@waku/interfaces";
+
+import { useFilterMessages, useStoreMessages } from "@waku/react";
+
+export const usePersistentNick = (): [
+  string,
+  React.Dispatch<React.SetStateAction<string>>
+] => {
+  const [nick, setNick] = useState<string>(() => {
+    const persistedNick = window.localStorage.getItem("nick");
+    return persistedNick !== null ? persistedNick : generate();
+  });
+  useEffect(() => {
+    localStorage.setItem("nick", nick);
+  }, [nick]);
+
+  return [nick, setNick];
+};
+
+type UseMessagesParams = {
+  node: undefined | LightNode;
+  decoder: undefined | Decoder;
+  options: StoreQueryOptions;
+};
+
+type UseMessagesResult = [Message[], (v: Message[]) => void];
+
+export const useMessages = (params: UseMessagesParams): UseMessagesResult => {
+  const { messages: newMessages } = useFilterMessages(params);
+  const { messages: storedMessages } = useStoreMessages(params);
+  const [localMessages, setLocalMessages] = useState<Message[]>([]);
+
+  const pushMessages = (msgs: Message[]) => {
+    if (!msgs || !msgs.length) {
+      return;
+    }
+    setLocalMessages((prev) => [...prev, ...msgs]);
+  };
+
+  const allMessages = React.useMemo((): Message[] => {
+    return [...storedMessages, ...newMessages]
+      .map(Message.fromWakuMessage)
+      .concat(localMessages)
+      .filter((v): v is Message => !!v)
+      .sort(
+        (left, right) => left.timestamp.getTime() - right.timestamp.getTime()
+      );
+  }, [storedMessages, newMessages, localMessages]);
+
+  return [allMessages, pushMessages];
+};
+
+export const useNodePeers = (node: undefined | LightNode) => {
```

Comment:
I am wondering if it worth including into `@waku/react`?
@fryorcraken

Reply:
Add a comment for developer to ignore it or just remove it from UI and keep console.log to observe experimentation with new Peer Exchange.